features/base/participants/functions.js

```diff
@@ -1,22 +1,46 @@
 /**
  * Returns local participant from Redux state.
  *
- * @param {Function} participantsOrGetState - Either the
+ * @param {(Function|Participant[])} participantsOrGetState - Either the
  * features/base/participants Redux state or Redux's getState function to be
  * used to retrieve the features/base/participants state.
- * @private
  * @returns {(Participant|undefined)}
  */
 export function getLocalParticipant(participantsOrGetState) {
-    let participants;
+    const participants = _getParticipantsArray(participantsOrGetState);
 
-    if (typeof participantsOrGetState === 'function') {
-        const getState = participantsOrGetState;
+    return participants.find(p => p.local);
+}
+
+/**
+ * Returns participant by id from Redux state.
+ *
+ * @param {(Function|Participant[])} participantsOrGetState - Either the
+ * features/base/participants Redux state or Redux's getState function to be
+ * used to retrieve the features/base/participants state.
+ * @param {string} id - Participant ID.
+ * @private
+ * @returns {(Participant|undefined)}
+ */
+export function getParticipant(participantsOrGetState, id) {
+    const participants = _getParticipantsArray(participantsOrGetState);
+
+    return participants.find(p => p.id === id);
```

Comment:
Given that getLocalParticipant was introduced and immediately reused, I believe you should've reused getParticipant immediately upon its introduction. I grepped the source code for `p => p.id === id` and `p => p.local` and discovered these source code snippets in multiple locations. According to our coding style, duplication is forbidden. In the case of these source code snippets we're improving readability in addition to reducing maintenance because getParticipant (or getParticipantById) and getLocalParticipant are easier to comprehend. (I'm taking care of this myself again.)